application/client.core/src/app/environment/components/views/chart/service.data.ts

```diff
@@ -487,7 +487,8 @@ export class ServiceData {
     }
 
     private _getValidNumberValue(val: string): number | undefined {
-        const value: number = parseInt(val, 10);
+        const regex = /^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$/;
+        const value: number = regex.test(val) ? parseFloat(val) : parseInt(val, 10);
```

Comment:
I am not sure we need regexp here:
1. it's not okay to construct it each time, if it's const you can just create in top of module
```
const FLOAT_TEST = /..../gi;
```
and you will not construct it each time with calling of `_getValidNumberValue`

2. But I think you can get rid of regexp here.
```
const floatOrInt: number = parseFloat(str);
if (floatOrInt % 1 == 0) {
     // I am INTEGER
} else {
     // I am FLOAT
}
```
This solution should be tested for sure

Reply:
Using of RegExp would be more expensive in any way. So, better avoid it